### PR TITLE
Simplify navbar code further.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

File: src/EditorFeatures/Core/Extensibility/NavigationBar/AbstractEditorNavigationBarItemService.cs [hunk 1]
```diff
@@ -77,30 +84,16 @@ public Task<bool> TryNavigateToItemAsync(Document document, NavigationBarItem it
...
-                navigationService.TryNavigateToPosition(workspace, documentToNavigate.Id, navigationPoint.Position, navigationPoint.VirtualSpaces, options: null, cancellationToken);
-            }
-            else
-            {
-                var notificationService = workspace.Services.GetRequiredService<INotificationService>();
-                notificationService.SendNotification(EditorFeaturesResources.The_definition_of_the_object_is_hidden, severity: NotificationSeverity.Error);
-            }
+            // We only show items in gray when near that actually exist (i.e. are not meant for codegen).
+            // This will be all C# items, and only VB non-codegen items.
+            return ((WrappedNavigationBarItem)item).UnderlyingItem is SymbolItem;
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/Extensibility/NavigationBar/AbstractEditorNavigationBarItemService.cs:96]: this was just pulling the VB impl up to the VB/C# shared impl.  It's the same effective logic for both.

File: src/EditorFeatures/Core/Extensibility/NavigationBar/AbstractEditorNavigationBarItemService.cs [hunk 2]
```diff
@@ -77,30 +84,16 @@ public Task<bool> TryNavigateToItemAsync(Document document, NavigationBarItem it
...
-
-            if (navigationService.CanNavigateToPosition(workspace, documentToNavigate.Id, navigationPoint.Position, navigationPoint.VirtualSpaces, cancellationToken))
-            {
-                navigationService.TryNavigateToPosition(workspace, documentToNavigate.Id, navigationPoint.Position, navigationPoint.VirtualSpaces, options: null, cancellationToken);
-            }
-            else
-            {
-                var notificationService = workspace.Services.GetRequiredService<INotificationService>();
-                notificationService.SendNotification(EditorFeaturesResources.The_definition_of_the_object_is_hidden, severity: NotificationSeverity.Error);
-            }
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/Extensibility/NavigationBar/AbstractEditorNavigationBarItemService.cs:100]: had this code already.  So removed the duplicated method and had the caller just call into the otehr impl.

File: src/EditorFeatures/Core/Extensibility/NavigationBar/AbstractEditorNavigationBarItemService.cs [hunk 3]
```diff
@@ -43,11 +44,17 @@ public Task<bool> TryNavigateToItemAsync(Document document, NavigationBarItem it
             var workspace = document.Project.Solution.Workspace;
 
             var (documentId, position, virtualSpace) = await GetNavigationLocationAsync(document, item, symbolItem, textSnapshot, cancellationToken).ConfigureAwait(false);
-            var navigationService = workspace.Services.GetRequiredService<IDocumentNavigationService>();
 
             // Ensure we're back on the UI thread before either navigating or showing a failure message.
             await this.ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/Extensibility/NavigationBar/AbstractEditorNavigationBarItemService.cs:50]:
```suggestion
```

File: src/EditorFeatures/VisualBasic/NavigationBar/VisualBasicEditorNavigationBarItemService_CodeGeneration.vb [hunk 4]
```diff
@@ -38,7 +38,13 @@ Namespace Microsoft.CodeAnalysis.Editor.VisualBasic.NavigationBar
...
                 newDocument.Project.Solution.Workspace.ApplyDocumentChanges(newDocument, cancellationToken)
 
-                NavigateToVirtualTreePoint(newDocument.Project.Solution, navigationPoint, cancellationToken)
+                Dim solution = newDocument.Project.Solution
+                NavigateToPosition(
+                    solution.Workspace,
+                    solution.GetRequiredDocument(navigationPoint.Tree).Id,
+                    navigationPoint.Position,
+                    navigationPoint.VirtualSpaces,
+                    cancellationToken)
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/VisualBasic/NavigationBar/VisualBasicEditorNavigationBarItemService_CodeGeneration.vb:47]:
```suggestion
                    solution.Workspace, solution.GetRequiredDocument(navigationPoint.Tree).Id,
                    navigationPoint.Position, navigationPoint.VirtualSpaces, cancellationToken)
```

**Review by dibarbet — APPROVED** [review]

**Pull request merged**